src/main/java/com/nutomic/syncthingandroid/activities/MainActivity.java

```diff
@@ -332,6 +359,40 @@ private Dialog createRestartDialog(){
                 .create();
     }
 
+    public void showQrCodeDialog(String deviceId, Bitmap qrCode) {
+        mQrCodeDialog = createQrCodeDialog(createQrCodeView(deviceId, qrCode));
+        mQrCodeDialog.show();
+    }
+
+    private AlertDialog createQrCodeDialog(View view) {
+        return new AlertDialog.Builder(this)
+                .setTitle(R.string.device_id)
+                .setView(view)
+                .setPositiveButton(R.string.finish, null)
+                .create();
+    }
+
+    private View createQrCodeView(String deviceId, Bitmap qrCode) {
+        View qrCodeDialogView = this.getLayoutInflater().inflate(R.layout.dialog_qrcode, null);
+        TextView deviceIdTextView = (TextView) qrCodeDialogView.findViewById(R.id.device_id);
+        TextView shareDeviceIdTextView = (TextView) qrCodeDialogView.findViewById(R.id.actionShareId);
+        ImageView qrCodeImageView = (ImageView) qrCodeDialogView.findViewById(R.id.qrcode_image_view);
+
+        deviceIdTextView.setText(deviceId);
+        deviceIdTextView.setOnClickListener(v -> Util.copyDeviceId(this, deviceIdTextView.getText().toString()));
+        shareDeviceIdTextView.setOnClickListener(v -> shareDeviceId(deviceId));
+        qrCodeImageView.setImageBitmap(qrCode);
+
+        return qrCodeDialogView;
+    }
```

Comment:
I think splitting it into 3 methods actually makes it more complicated, because you have to figure out which one is actually used, and how (especially as the names are all really similar).